internal/ledgerutil/compare.go

```diff
@@ -164,24 +183,24 @@ func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count
 			// Add missing to output JSON file
 			diffRecord, err := newDiffRecord(namespace2, nil, snapshotRecord2)
 			if err != nil {
-				return 0, err
+				return 0, "", err
 			}
 			err = jsonOutputFile.addRecord(*diffRecord)
 			if err != nil {
-				return 0, err
+				return 0, "", err
 			}
 			namespace2, snapshotRecord2, err = snapshotReader2.Next()
 			if err != nil {
-				return 0, err
+				return 0, "", err
 			}
 		}
 	}
 
 	err = jsonOutputFile.close()
 	if err != nil {
-		return 0, err
+		return 0, "", err
 	}
-	return jsonOutputFile.count, nil
+	return jsonOutputFile.count, allRes, nil
```

Comment:
I don't think that we need to return the file name from here. This is not used anywhere anyways!

Reply:
This was done for the tests because they require the full file path